### PR TITLE
[bugfix] Fix `AttributeError` in flexible node allocation from the Slurm backend

### LINKED ISSUE #1258: AttributeError is raised in 'filternodes' method of slurm scheduler
The exception is raised when `job.sched_access` is a `SequenceView` object and subsequent statements try to append values to an immutable object.

### PR DESCRIPTION
**Pull request by teojgo**
* Fix `AttributeError` raised when trying to mutate a `SequenceView`
  object in the `filternodes` method of the slurm scheduler.

* Add a corresponding unittest.

Fixes #1258

### CODE REVIEW

File: reframe/core/schedulers/slurm.py [hunk 1]
```diff
@@ -231,19 +231,19 @@ def filternodes(self, job, nodes):
         # Collect options that restrict node selection
         options = job.sched_access + job.options
         if job.sched_partition:
-            options.append('--partition=%s' % job.sched_partition)
+            options += ['--partition=%s' % job.sched_partition]
```
Comment (ekouts) [inline reframe/core/schedulers/slurm.py:234]:
```suggestion
            options += [f'--partition={job.sched_partition}']
```
Should you change this to use f-strings?

File: reframe/core/schedulers/slurm.py [hunk 2]
```diff
@@ -231,19 +231,19 @@ def filternodes(self, job, nodes):
...
 
         if job.sched_account:
-            options.append('--account=%s' % job.sched_account)
+            options += ['--account=%s' % job.sched_account]
```
Comment (ekouts) [inline reframe/core/schedulers/slurm.py:237]: Same for all the options

**Comment by vkarak** [comment]
@teojgo I can't understand where this `SequenceView` object comes from.

File: reframe/core/schedulers/slurm.py [hunk 3]
```diff
@@ -231,19 +231,19 @@ def filternodes(self, job, nodes):
         # Collect options that restrict node selection
         options = job.sched_access + job.options
```
Comment (vkarak) [inline reframe/core/schedulers/slurm.py:232]: It's more correct conceptually if you change just this line:

```suggestion
        # Create a mutable list out of the SequenceView object of `job.sched_access`
        options = list(job.sched_access + job.options)
```

File: unittests/test_schedulers.py [hunk 4]
```diff
@@ -743,6 +744,16 @@ def test_sched_access_idle(self):
         self.prepare_job()
         assert self.testjob.num_tasks == 8
 
+    def test_sched_access_idle_sequence_view(self):
```
Comment (vkarak) [inline unittests/test_schedulers.py:747]: I don't think this is needed. Just change the `_sched_access` assignment in one of the `sched_access` tests.

**Comment by vkarak** [comment]
It seems that this unit test is not enough. It does not break if you revert the fix. Since I'm fixing the PR, I will have a look at it as well.

**Comment by teojgo** [comment]
> It seems that this unit test is not enough. It does not break if you revert the fix. Since I'm fixing the PR, I will have a look at it as well.

@varak yes, that's why I had the separate unit test because you need both `self.testjob._sched_access` and one of the other `self.test_job._sched...` in order to try to mutate the SequenceView.

**Comment by vkarak** [comment]
@teojgo Yes, just realized that. I will cherry pick that commit.

**Pull request merged**